tests/src/commonTest/kotlin/tests/VerificationTests.kt

```diff
@@ -99,6 +99,32 @@ class VerificationTests {
         assertEquals("Cannot verify MockFoo.newInt() as it has not been mocked", ex.message)
     }
 
+    @Test
+    fun testOrderedMultipleMocks() {
+        val foo1 = MockFoo<Bar>(mocker)
+        val foo2 = MockFoo<Bar>(mocker)
+        mocker.every { foo1.doInt(isAny()) } returns Unit
+        mocker.every { foo1.newString() } returns ""
+        mocker.every { foo2.doInt(isAny()) } returns Unit
+        mocker.every { foo2.newString() } returns ""
+
+        foo1.doInt(42)
+        foo2.doInt(42)
+        foo1.newString()
+        foo2.newString()
+
+        val ex = assertFailsWith<AssertionError> {
+            mocker.verify(inOrder = true) {
+                foo1.doInt(42)
+                foo2.doInt(42)
+                // Same method but different receiver order
+                foo2.newString()
+                foo1.newString()
+            }
+        }
+        assertEquals("Expected a call to MockFoo.newString(), but was a call to MockFoo.newString()", ex.message)
```

Comment:
Note that the message does NOT convey the problem cause. We may want to change the rendering of the receiver when both are matching (maybe with something like `MockFoo$142b15ac.newString()` based on hashCode() ?).

Reply:
OK, I'll add that.